### PR TITLE
Introduce SwiftLint tool to enforce Swift style and conventions

### PR DESCRIPTION
**Pull request by pixyzehn**
Introduced SwiftLint and Fixed warnings and errors.
Now `Marathon` doesn't have xcodeproj, so we just run SwiftLint in CI or each local env.

In `type_body_length` and `file_length `, I adjusted the boundary value based on the following warnings and errors
```sh
Marathon/Tests/MarathonTests/MarathonTests.swift:13:1: error: Type Body Length Violation: Type body should span 350 lines or less excluding comments and whitespace: currently spans 388 lines (type_body_length)
...

Marathon/Tests/MarathonTests/MarathonTests.swift:784: warning: File Line Length Violation: File should contain 400 lines or less: currently contains 784 (file_length)
...
```

Ref:
https://github.com/krzysztofzablocki/Sourcery/blob/master/.swiftlint.yml
https://www.buddybuild.com/blog/customizing-the-build-process

### CODE REVIEW

**Comment by aksswami** [comment]
We can also use [SwiftFormat](https://github.com/nicklockwood/SwiftFormat). The benefit of having `SwiftFormat` over `SwiftLint` is it automate the process of maintaining one syntax styling. Whereas in `SwiftLint` it will generate all the warning and developer have to resolve these. 

I would like a solution where a tool does this work for me, automatically.

**Comment by pixyzehn** [comment]
@aksswami I don't know the details, but you may use the same things by using `swiftlint autocorrect`.
https://github.com/realm/SwiftLint/tree/480acd77f5878374eb7a4e8b122c61bc34a6d88b#format-on-save-xcode-plugin

File: Sources/MarathonCore/Package.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ public struct Package {
 }
 
 extension Package: Equatable {
-    public static func ==(lhs: Package, rhs: Package) -> Bool {
+    public static func == (lhs: Package, rhs: Package) -> Bool {
```
Comment (JohnSundell) [inline Sources/MarathonCore/Package.swift]: Is this rule disablable? It doesn't make sense to me as functions are always defined `foo()` without a space.

File: Tests/MarathonTests/Assert.swift [hunk 2]
```diff
@@ -47,9 +47,9 @@ public func assert<T, E: Error>(at file: StaticString = #file,
  *  - closure: The closure that should thrown an error
  */
 public func assertErrorThrown<T, E: Error>(at file: StaticString = #file,
-                              line: UInt = #line,
-                              _ errorExpression: @autoclosure () -> E,
-                              by closure: () throws -> T) where E: Equatable {
+                                           line: UInt = #line,
```
Comment (JohnSundell) [inline Tests/MarathonTests/Assert.swift]: Nice 👍

File: .swiftlint.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - line_length
+  - force_try
+  - force_cast
+
+included:
+  - Sources
+  - Tests
+
+type_body_length:
+  - 700 #warning
```
Comment (JohnSundell) [inline .swiftlint.yml]: What measurement is this in?

File: Sources/MarathonCore/Package.swift (hunk 1 above)
Comment (pixyzehn) [inline Sources/MarathonCore/Package.swift]: It causes the warning like `warning: Operator Function Whitespace Violation: Operators should be surrounded by a single whitespace when defining them. (operator_whitespace)`.
We can disable it. What do you think? For me, it makes sense based on the [Advanced Operators](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/AdvancedOperators.html) section.

File: .swiftlint.yml (hunk 3 above)
Comment (pixyzehn) [inline .swiftlint.yml]: I referred to https://github.com/krzysztofzablocki/Sourcery/blob/master/.swiftlint.yml#L26.
If there is no rule about `type_body_length ` and `file_length`, we have an error.

```console
error: Type Body Length Violation: Type body should span 350 lines or less excluding comments and whitespace: currently spans 388 lines (type_body_length)
warning: File Line Length Violation: File should contain 400 lines or less: currently contains 784 (file_length)
```

How about the following rule?
```yaml
type_body_length:
  - 400 #warning

file_length:
  - 800 #warning
```
Or we can disable them by using `swiftlint:disable ***`

**Comment by pixyzehn** [comment]
@JohnSundell Could you check my update?

File: Sources/MarathonCore/Package.swift (hunk 1 above)
Comment (JohnSundell) [inline Sources/MarathonCore/Package.swift]: I'd love if we could disable it 😇

Comment (pixyzehn) [inline Sources/MarathonCore/Package.swift]: @JohnSundell No problem:) I updated.

**Comment by JohnSundell** [comment]
We could just execute `swiftlint` in our Travis & BuddyBuild scripts, right?

**Comment by pixyzehn** [comment]
@JohnSundell No problem:) Yes, we want! 

> We could just execute `swiftlint` in our Travis & BuddyBuild scripts, right?

That's right! But I think it's better to set it in other PR for now if possible. It needs some adjustments like installing SwiftLint by homebrew in Travis...

**Comment by JohnSundell** [comment]
@pixyzehn Sounds great! 👍 Do you wanna create an issue for tracking adding this to CI? 😉

**Comment by pixyzehn** [comment]
@JohnSundell Yes, I want! I'll investigate it👍 Thanks again!
